src/gap-skipper.js

```diff
@@ -0,0 +1,85 @@
+import Ranges from './ranges';
+
+let seeking = false;
+let consecutiveUpdates = 0;
+let timer = null;
+let playerState;
+let lastRecordedTime;
+let adaptiveSeeking;
+let player;
+let tech;
+let gapSkipper;
+
+gapSkipper = function(options) {
+  player = this;
+  tech = options.tech;
+
+  // Allows us to mimic a waiting event in chrome
+  player.on('timeupdate', function() {
+    if (player.paused()) {
+      return;
+    }
+    let currentTime = player.currentTime();
+
+    if (consecutiveUpdates === 5 && currentTime === lastRecordedTime) {
+      // trigger waiting
+      if (playerState !== 'waiting') {
+        consecutiveUpdates = 0;
+        playerState = 'waiting';
+        tech.trigger('adaptive-seeking');
```

Comment:
we should probably trigger this on the new gap-skipper object rather than on the tech directly